kai/llm_interfacing/model_provider.py

```diff
@@ -209,62 +206,32 @@ def __init__(
     def invoke(
         self,
         input: LanguageModelInput,
+        cache_path_resolver: CachePathResolver,
```

Comment:
Can we move this to a kwarg that we strip out so that the interface for invoke is the same for the base_model

Reply:
@shawn-hurley I made this explicit so agent writers know they need to pass this, if they forget, cache won't be generated. Is there any specific reason you wanted to keep the api untouched?